src/hotspot/os/posix/perfMemory_posix.cpp

```diff
@@ -104,21 +104,15 @@ static void save_memory_to_file(char* addr, size_t size) {
   } else {
     ssize_t result;
 
-    for (size_t remaining = size; remaining > 0;) {
-
-      result = os::write(fd, addr, remaining);
-      if (result == OS_ERR) {
-        if (PrintMiscellaneous && Verbose) {
-          warning("Could not write Perfdata save file: %s: %s\n",
-                  destfile, os::strerror(errno));
-        }
-        break;
+    result = os::write(fd, addr, size);
+    if (result == OS_ERR) {
+      if (PrintMiscellaneous && Verbose) {
```

Comment:
It's not really part of this issue but since the line is changed, can you change it to unconditionally
   log_info(os)("Could not write...);
And remove PrintMiscellaneous & Verbose.

Reply:
Done.